dev/com.ibm.ws.microprofile.faulttolerance.cdi/src/com/ibm/ws/microprofile/faulttolerance/cdi/FaultToleranceCDIComponent.java

```diff
@@ -67,22 +69,18 @@ protected void deactivate() {
      * <p>
      * The provider may change if features are updated
      */
-    public static MetricRecorderProvider getMetricProvider() {
-        return metricProvider;
+    public static List<MetricRecorderProvider> getMetricProviders() {
+        return metricProviders;
     }
 
     // Require a provider, update dynamically if a new one becomes available
-    @Reference(policy = DYNAMIC, policyOption = GREEDY, cardinality = MANDATORY)
+    @Reference(policy = DYNAMIC, policyOption = GREEDY, cardinality = AT_LEAST_ONE)
     protected void setProvider(MetricRecorderProvider provider) {
-        metricProvider = provider;
+        metricProviders.add(provider);
     }
 
     protected void unsetProvider(MetricRecorderProvider provider) {
-        // Unless we're shutting down, we expect this to be called just *after* setProvider has been called with the new provider
-        // so don't unset the field unconditionally.
-        if (metricProvider == provider) {
-            metricProvider = null;
-        }
+        metricProviders.remove(provider);
     }
```

Comment:
In the original code the cardinality is `MANDATORY` because there should always be a provider because the same bundle registers a dummy provider. If the dummy and the real provider are both registered, we'll get the real one because it has a higher ranking.

In the new code, as it's currently written, we'll return either a list containing the dummy provider, or a list containing some real providers and the dummy provider.

In the latter case, we don't want to include the dummy provider. It should just be the backup if there are no other providers.